code/network/multimsgs.cpp

```diff
@@ -9114,3 +9115,133 @@ void process_sexp_packet(ubyte *data, header *hinfo)
 
 	sexp_packet_received(received_packet, num_ubytes);
 }
+
+// TODO, calculate how many we can send.
+constexpr ubyte MAX_TURRET_TRACK_CHANGES = 32;
+constexpr int SIZE_LOCATION_IN_PACKET = 7;
+
+SCP_vector<int> send_turret_tracking_packet(ushort parent_sig, SCP_vector<int>& subsys_indexes) 
+{
+	Assertion(MULTIPLAYER_MASTER, "A non-server is trying to send a turret tracking packet. This is a coder mistake, please report!");
+
+	SCP_vector<int> overflow_list;
+
+	// somehow trying to send for an an invalid or non-network object
+	if (parent_sig == 0) {
+		// returning an empty list will signal the manager to clear its entries
+		return overflow_list;
+	}
+
+	auto temp = multi_get_network_object(parent_sig);
+
+	// this means that the ship may have been destroyed before being able to send the packet.  Ignore.
+	if (temp == nullptr){
+		// returning an empty list will signal the manager to clear its entries
+		return overflow_list;
+	}
+
+	ship* shipp = &Ships[temp->instance];
+
+	// Required for all send functions
+	ubyte data[MAX_PACKET_SIZE];
+	int packet_size = 0;
+
+	BUILD_HEADER(TURRET_TRACK);
+
+	// send the time that this happened.
+	int time_out = Multi_Timing_Info.get_current_time();
+	ADD_INT(time_out);
+	ADD_USHORT(parent_sig);
+
+	ubyte last_index = (subsys_indexes.size() < MAX_TURRET_TRACK_CHANGES) ? static_cast<ubyte>(subsys_indexes.size()) : MAX_TURRET_TRACK_CHANGES;
+
+	// this is not usually done, but if the number of subsystems actually ends up being different than limit,
+	// this ubyte in the packet will get overwritten at the end of this function.
+	ADD_DATA(last_index);
+
+	ubyte last_index_record = last_index;
+
+	ubyte count = 0;
+	for (ubyte i = 0; i < subsys_indexes.size(); i++){
+
+		// pre count is now being switched to a limiter to ensure that we do not exceed MAX_TURRET_TRACK_CHANGES
+		if (i < last_index) {
+			auto ssp = ship_get_indexed_subsys( shipp, subsys_indexes[i]);
+
+			// make sure the subsytem is valid.  If not, just continue.
+			if (ssp != nullptr){
+				// Add the data for the valid entry
+				ADD_SHORT( static_cast<short>(subsys_indexes[i]));
+
+				ushort target_netsig = (ssp->turret_enemy_objnum > -1) ? Objects[ssp->turret_enemy_objnum].net_signature : 0;
+				ADD_USHORT(target_netsig);
+
+				packet_size += multi_pack_turret_angles(data + packet_size, ssp);
+				++count;
+			} else {
+				// if we don't have a valid
+				++last_index;
+			}
+
+		} else {
+			overflow_list.push_back(subsys_indexes[i]);
+		}
+	}
+
+	// and here's where the correction is done.  This should be rare.
+	if (count != last_index_record) {
+		// this is a little hacky, but best to work through the serialization macros to rewrite the data.
+		int packet_size_record = packet_size;
+		packet_size = SIZE_LOCATION_IN_PACKET;
+		ADD_DATA(count);
+		packet_size = packet_size_record;
+	}
+
```

Comment:
You should change that count > 0 here before sending and avoid wasting bandwidth on useless packets.